lib/provider.go

```diff
@@ -312,24 +312,18 @@ func (p *Provider) roleSessionName() string {
 	return fmt.Sprintf("%d", time.Now().UTC().UnixNano())
 }
 
-// GetRoleARN uses p to establish temporary credentials then calls
-// lib.GetRoleARN with them to get the role's ARN
-func (p *Provider) GetRoleARN() (string, error) {
-	creds, err := p.getSamlSessionCreds()
-	if err != nil {
-		return "", err
+// GetRoleARN uses temporary credentials to call AWS's get-caller-identity and
+// returns the assumed role's name and ARN.
+func (p *Provider) GetRoleARN(creds credentials.Value) (string, error) {
```

Comment:
`lib.GetRoleARN` was added so we could pass the credentials in without breaking API compatibility of `Provider#GetRoleARN`. I realize the API is gross and probably nobody imports it externally, but we can't be sure.

How about just adding a new function and method `GetRoleARNWithRegion`? I realize this contributes to API sprawl, but we can tackle that in v3.0

Reply:
done (and restored the old `lib.GetRoleARN`)!